compiler/rustc_ty_utils/src/opaque_types.rs

```diff
@@ -159,6 +246,28 @@ impl<'tcx> TypeVisitor<TyCtxt<'tcx>> for OpaqueTypeCollector<'tcx> {
                         }
                     }
                 }
+
+                // Normalize trivial projections.
+                let mut infcx = self.tcx.infer_ctxt();
+                let infcx = infcx.build();
+                let t = if t.has_escaping_bound_vars() {
+                    let (t, _mapped_regions, _mapped_types, _mapped_consts) =
+                        rustc_trait_selection::traits::project::BoundVarReplacer::replace_bound_vars(
+                            &infcx,
+                            &mut self.universes,
+                            t,
+                        );
+                    t
+                } else {
+                    t
+                };
+                let ocx = ObligationCtxt::new(&infcx);
+                let cause = ObligationCause::dummy_with_span(self.span());
+                let normalized = ocx.normalize(&cause, self.tcx.param_env(self.item), t);
+                trace!(?normalized);
```

Comment:
you need to select_where_possible here i think

Reply:
Not anymore https://github.com/rust-lang/rust/pull/112652/commits/6cd8a832f35373277e62378936b36201c5567193 

don't need to select if you aren't normalizing *taps head*